core/lib/types/src/protocol_version.rs

```diff
@@ -462,6 +367,164 @@ impl TryFrom<Log> for ProtocolUpgrade {
     }
 }
 
+pub fn decode_set_chain_id_event(
+    event: Log,
+) -> Result<(ProtocolVersionId, ProtocolUpgradeTx), crate::ethabi::Error> {
+    let transaction_param_type = ParamType::Tuple(vec![
```

Comment:
Nit: This definition is duplicated from `impl TryFrom<Log> for ProtocolUpgrade` above; maybe, it makes sense to extract it to a function?